smartdashboard/views.py

```diff
@@ -341,3 +345,467 @@ def __init__(
+            "timestamp", var_name="Metric", value_name="Memory (GB)"
+        )
+
+        if self.chart is None or self.sampling:
+            chart = (
+                alt.Chart(dframe)
+                .mark_line()
+                .encode(
+                    x=alt.X(
+                        "timestamp:Q",
+                        axis=alt.Axis(title="Timestep in seconds", labelAngle=0),
+                    ),
+                    y=alt.Y("Memory (GB):Q", axis=alt.Axis(title="Memory in GB")),
+                    color=alt.Color(  # type: ignore[no-untyped-call]
+                        "Metric:N", scale=alt.Scale(scheme="category10"), title="Legend"
+                    ),
+                    tooltip=["timestamp:Q", "Metric:N", "Memory (GB):Q"],
+                )
+                .interactive()
+                .properties(
+                    height=500, title=alt.TitleParams("Memory Usage", anchor="middle")
+                )
+                .configure_legend(orient="bottom")
+            )
+
+            self.graph_element.altair_chart(
+                chart, use_container_width=True, theme="streamlit"
+            )
+            self.chart = chart
+
+        else:
+            self.graph_element.add_rows(dframe)
+
+    def _update_table(self, dframe: pd.DataFrame) -> None:
+        """Update memory table for selected shard
+
+        :param dframe: DataFrame with memory data
+        :type dframe: pandas.DataFrame
+        """
+
+        dframe = dframe.drop(columns=["timestamp"])
+        self.table_element.dataframe(
+            dframe.tail(1), use_container_width=True, hide_index=True
+        )
+
+
+class ClientView(DualView):
+    """View class for client section of the Database Telemetry page"""
+
+    @property
+    def files(self) -> Files:
+        """Returns a tuple of the files used for telemetry"""
+        if self.shard is not None:
+            return Files(self.shard.client_count_file, self.shard.client_file)
+
+        return Files("", "")
+
+    @property
+    def message(self) -> str:
+        """Returns the error message if files cannot be found"""
+        if self.shard is not None:
+            return f"Client information could not be found for {self.shard.name}"
+
+        return ""
+
+    @property
+    def columns(self) -> t.List[str]:
+        """Returns columns for the graph dataframe"""
+        return ["timestamp", "num_clients"]
+
+    def enable_export_button(self) -> None:
+        """Create an export data button"""
+        if self.shard is not None and self.telemetry:
+            try:
+                self.export_button.download_button(
+                    label="Export Data",
+                    data=self._get_data_file(),
+                    file_name=f"{self.shard.name}-clients.csv",
+                    mime="text/csv",
+                    key="clients",
+                )
+            except FileNotFoundError:
+                self.export_button.empty()
```

Comment:
Which method exactly are we expecting to raise a `FileNotFoundError` here?

If it `self._get_data_file`, from what I can tell, it will already handle that on its own.

If it is `download_button` itself, I would expect this error check would need to go in some sort of callback for the button to execute on click, but I might just be misunderstanding the streamlit API.

Reply:
Nope you're right! I've removed the `FileNotFoundError`s and added the `self.export_button.empty()` in `self._get_data_file` when it check the file.